task_2/sergii.garanenko/src/main/java/com/greyslon/servlets/servlets/GetBooksByAuthorServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.greyslon.servlets.servlets;
+
+import com.greyslon.servlets.container.ObjectsContainer;
+import com.greyslon.servlets.models.Book;
+import com.greyslon.servlets.repositories.BookRepository;
+import com.greyslon.servlets.utils.HtmlContent;
+import com.greyslon.servlets.utils.HtmlHelper;
+import com.greyslon.servlets.utils.QueryParamGetter;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URLDecoder;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/books/search")
+public class GetBooksByAuthorServlet extends HttpServlet {
+
+  private QueryParamGetter paramGetter;
+  private BookRepository bookRepository;
+  private HtmlContent htmlContent;
+  private HtmlHelper htmlHelper;
+
+  public GetBooksByAuthorServlet() {
+    bookRepository = ObjectsContainer.getObject(BookRepository.class);
+    htmlContent = ObjectsContainer.getObject(HtmlContent.class);
+    htmlHelper = ObjectsContainer.getObject(HtmlHelper.class);
+    paramGetter = ObjectsContainer.getObject(QueryParamGetter.class);
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+    resp.setContentType("text/html;charset=UTF-8");
+    PrintWriter writer = resp.getWriter();
+    String queryString = URLDecoder.decode(req.getQueryString(), "UTF-8");
+    Optional<String> author = paramGetter.getParam(queryString, "author");
+    if (author.isPresent()) {
+      Set<Book> books = bookRepository.getBooks(author.get());
+      writer.write(htmlContent.renderNavigation());
+      if (books.isEmpty()) {
```

Comment:
It is better to reduce internal **ifs**. Moving that **if-else** block in a separate private method will be better.